### PR TITLE
 Moved beam_size from constructor to search fct.

### LINKED ISSUE #500: Move `beam_size` from constructor to `search` in BeamSearch.
This should be very simple to do and would add a lot of simplicity.

### PR DESCRIPTION
**Pull request by Thrandis**
Fixes #500

### CODE REVIEW

File: blocks/search.py [hunk 1]
```diff
@@ -277,6 +276,7 @@ def search(self, input_values, eol_symbol, max_length,
             negative log-likelihood.
 
         """
+        self.beam_size = input_values[input_values.keys()[0]].shape[1]
```
Comment (dmitriy-serdyuk) [inline blocks/search.py]: It's better to provide beam size as an additional argument to `compute_initial_states`, anyway you don't use it anywhere else.

File: .travis.yml [hunk 2]
```diff
@@ -52,7 +52,8 @@ script:
   - |
       if [[ $TESTS == 'blocks-examples' ]]; then
         cd $TRAVIS_BUILD_DIR
-        git clone https://github.com/mila-udem/blocks-examples.git
+        #git clone https://github.com/mila-udem/blocks-examples.git
+        git clone https://github.com/Thrandis/blocks-examples.git
```
Comment (dmitriy-serdyuk) [inline .travis.yml]: You need to clone your branch with corresponding changes (the syntax is `git clone -b my-branch git@github.com:user/myproject.git`).

File: blocks/search.py (hunk 1 above)
Comment (rizar) [inline blocks/search.py]: I am afraid this is not a generic solution. A sequence generator can simply have no sequential inputs, consider an encoder-decoder model.

However, after #719 is solved the beam size seems to be not necessary at all!

**Comment by Thrandis** [comment]
@rizar good point! I'll move to #719 then!

**Pull request closed without merging**